### PR TITLE
Controlling more node types

### PR DESCRIPTION
**Pull request by SouthEndMusic**
Fixes https://github.com/Deltares/Ribasim/issues/398.

### CODE REVIEW

**Comment by SouthEndMusic** [comment]
Here is the draft for parsing the static data for control, currently only applied to `Pump`.

File: core/src/create.jl [hunk 1]
```diff
@@ -1,3 +1,73 @@
...
+    db::DB,
+    nodetype::String,
+    defaults::NamedTuple,
+)::NamedTuple
+    static_type = eltype(static)
+    columnnames = collect(fieldnames(static_type))
+    mask = [symb ∉ [:node_id, :control_state] for symb in columnnames]
+    columnnames = columnnames[mask]
+    columntypes = collect(fieldtypes(static_type))[mask]
+    vals = Vector{Any}()
```
[the remark this answers is not in the document]
Comment (SouthEndMusic) [reply core/src/create.jl:12]: Didn't know that was an option (:

[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply core/src/create.jl]: Hah, missed that one

File: core/src/create.jl [hunk 2]
```diff
@@ -1,3 +1,82 @@
...
+
+    # Node id of the node being processed
+    node_id = node_ids[1]
+
+    # Index in the output vectors for this node ID
+    node_idx = 1
+
+    for row in static
+
+        # I hope the alignment of values goes well, assumes ordered node IDs
```
Comment (visr) [inline core/src/create.jl:53]: `load_structvector` guarantees that it is ordered by node_id

**Review by visr — APPROVED** [review]

**Pull request merged**